weld/llvm.rs

```diff
@@ -3453,84 +3453,75 @@ impl LlvmGenerator {
-                                    for s in b.statements.iter() {
-                                        self.gen_statement(s, keyfunc, keyfunc_ctx)?
+                                    key_ll_sym = key_ll_symc;
+                                    // Generate prelude code
+                                    let keyfunc_ctx = &mut FunctionContext::new(false);
+                                    keyfunc_ctx.alloca_code.add(format!("fn.entry:"));
+                                    for (arg, ty) in keyfunc.locals.iter() {
+                                        let arg_str = llvm_symbol(&arg);
+                                        let ty_str = self.llvm_type(&ty)?;
+                                        keyfunc_ctx.add_alloca(&arg_str, &ty_str)?;
                                     }
+                                    keyfunc_ctx.code.add(format!("br label %b.b{}", keyfunc.blocks[0].id));
+                                    let end_block_index = keyfunc.blocks.len() - 1;
+                                    for (i, b) in keyfunc.blocks.iter().enumerate() {
+                                        keyfunc_ctx.code.add(format!("b.b{}:", b.id));
+                                        for s in b.statements.iter() {
+                                            self.gen_statement(s, keyfunc, keyfunc_ctx)?
+                                        }
+                                        if (i != end_block_index) {
+                                            if let Terminator::Branch { ref cond, on_true, on_false} = b.terminator {
+                                                keyfunc_ctx.code.add(format!("; {}", b.terminator));
+                                                if self.trace_run {
+                                                    self.gen_puts(&format!("  {}", b.terminator), ctx);
+                                                }
+                                                let cond_tmp = self.gen_load_var(llvm_symbol(cond).as_str(), "i1", keyfunc_ctx)?;
+                                                keyfunc_ctx.code.add(format!("br i1 {}, label %b.b{}, label %b.b{}", cond_tmp, on_true, on_false));
+                                            } else if let Terminator::JumpBlock(block) = b.terminator {
+                                                keyfunc_ctx.code.add(format!("; {}", b.terminator));
+                                                if self.trace_run {
+                                                    self.gen_puts(&format!("  {}", b.terminator), ctx);
+                                                }
+                                                keyfunc_ctx.code.add(format!("br label %b.b{}", block));
+                                            } else if let Terminator::ProgramReturn(_) = b.terminator {
+                                            } else {
+                                                return weld_err!("Can't have intermediate return in sort key function");
```

Comment:
change this message to "Can't have terminator other than Branch or JumpBlock in sort key function"

Reply:
addressed.